opm/output/data/Wells.hpp

```diff
@@ -153,12 +153,20 @@ namespace Opm {
     struct Segment {
         Rates rates;
         double pressure;
+        double pressure_drop;
```

Comment:
Could we have a small struct: 
```C++
struct SegmentPressures {
     double pressure_drop;
     ....
};
```

Reply:
> Could we have a small struct:

I don't think we get enough benefit from that since we'll anyway have to access individual structure members to assign and read the values.  I think I'd rather have something slightly more involved
```C++
class SegmentPressure {
public:
    enum class Value : std::size_t {
        Pressure, PDrop, PDropHydrostatic, PDropAccel, PDropFriction,
    };

    double& operator[](const Value i)
    {
        return this->value_[this->index(i)];
    }

    double operator[](const Value i) const
    {
        return this->value_[this->index(i)];
    }

private:
    std::array<double, numvals> values_;

    std::size_t index(const Value ix) const
    {
        return static_cast<std::size_t>(ix);
    }
};
```
This would turn the [assignments]() into something like
```C++
segres.pressure[Value::Pressure] = 123.456;
```
and we could have a generic evaluation function of the form
```C++
template <SegmentPressure::Value ix>
double segpress(const fn_args& args)
{
    // … checks …
    return { segment.pressure[ix], measure::pressure };
}
```
which would be put into the summary table using code like
```C++
{ "SPR", segpress<SegmentPressure::Pressure> },
{ "SPRD", segpress<SegmentPressure::PDrop> },
{ "SPRDH", segpress<SegmentPressure::PDropHydrostatic> },
{ "SPRDF", segpress<SegmentPressure::PDropFriction> },
{ "SPRDA", segpress<SegmentPressure::PDropAccel> },
```

Reply:
> I don't think we get enough benefit from that since we'll anyway have to access individual structure members to assign and read the values. I think I'd rather have something slightly more involved

Thanks @bska , I will take your suggestion. This also makes the next comment easier to address.